alternator/conditions.cc

```diff
@@ -234,19 +237,24 @@ static bool check_LT(const rjson::value* v1, const rjson::value& v2) {
         return false;
     }
     if (kv1.name == "N") {
-        return unwrap_number(*v1, "LT operator") < unwrap_number(v2, "LT operator");
+        return cmp(unwrap_number(*v1, cmp.diagnostic()), unwrap_number(v2, cmp.diagnostic()));
     }
     if (kv1.name == "S") {
-        return std::string_view(kv1.value.GetString(), kv1.value.GetStringLength())
-                < std::string_view(kv2.value.GetString(), kv2.value.GetStringLength());
+        return cmp(std::string_view(kv1.value.GetString(), kv1.value.GetStringLength()),
+                   std::string_view(kv2.value.GetString(), kv2.value.GetStringLength()));
     }
     if (kv1.name == "B") {
-        return base64_decode(kv1.value) < base64_decode(kv2.value);
+        return cmp(base64_decode(kv1.value), base64_decode(kv2.value));
     }
-    clogger.error("check_LT panic: LHS type equals RHS type, but one is in {N,S,B} while the other isn't");
+    clogger.error("check_compare panic: LHS type equals RHS type, but one is in {N,S,B} while the other isn't");
     return false;
 }
 
+struct cmp_lt {
+    template <typename T> bool operator()(const T& lhs, const T& rhs) const { return lhs < rhs; }
+    const char* diagnostic() const { return "LT operator"; }
```

Comment:
Nitpic - this could have been a string constant instead of a function to make it clear it's just a constant. But this is also ok.